Source/WebKit/Shared/Extensions/_WKWebExtensionLocalization.mm

```diff
@@ -0,0 +1,334 @@
+/*
+ * Copyright (C) 2023 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#if !__has_feature(objc_arc)
+#error This file requires ARC. Add the "-fobjc-arc" compiler flag for this file.
+#endif
+
+#import "config.h"
+#import "_WKWebExtensionLocalization.h"
+
+#import "CocoaHelpers.h"
+#import "Logging.h"
+
+NSString * const WKLocalizationDictionaryMessageKey = @"message";
+NSString * const WKLocalizationDictionaryDescriptionKey = @"description";
+NSString * const WKLocalizationDictionaryPlaceholdersKey = @"placeholders";
+NSString * const WKLocalizationDictionaryJSONPathFormat = @"_locales/%@/messages.json";
+
+NSString * const WKPredefinedMessageUILocale = @"@@ui_locale";
+NSString * const WKPredefinedMessageLanguageDirection = @"@@bidi_dir";
+NSString * const WKPredefinedMessageLanguageDirectionReversed = @"@@bidi_reversed_dir";
+NSString * const WKPredefinedMessageTextLeadingEdge = @"@@bidi_start_edge";
+NSString * const WKPredefinedMessageTextTrailingEdge = @"@@bidi_end_edge";
+NSString * const WKPredefinedMessageExtensionID = @"@@extension_id";
+
+NSString * const WKPredefinedMessageValueLeftToRight = @"ltr";
+NSString * const WKPredefinedMessageValueRightToLeft = @"rtl";
+NSString * const WKPredefinedMessageValueTextEdgeLeft = @"left";
+NSString * const WKPredefinedMessageValueTextEdgeRight = @"right";
```

Comment:
Most of these are only used in this file, they should be `static` then and drop the `WK` prefix.

Reply:
I kept the first three as with the WK since I plan on using them for testing, but changed the rest